applications/callflow/src/module/cf_dynamic_cid.erl

```diff
@@ -43,9 +48,110 @@
                          )
          }).
 
+
 %%--------------------------------------------------------------------
 %% @public
 %% @doc
+%% Entry point for this module, based on the payload will either
+%% connect a caller to check_voicemail or compose_voicemail.
+%% @end
+%%--------------------------------------------------------------------
+-spec handle(wh_json:object(), whapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    case wh_json:get_value(<<"action">>, Data) of
+        <<"manual">> ->
+            lager:info("user must manually enter on keypad the caller id for this call"),
+	    handle_manual(Data, Call);
+        <<"list">> ->
+            lager:info("user is choosing a caller id for this call from a range of options"),
+	    handle_list(Data, Call);
+        _ ->
+	    lager:info("user must manually enter on keypad the caller id for this call"),
+	    handle_manual(Data, Call)  
+    end.
+
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Entry point for this module
+%% @end
+%%--------------------------------------------------------------------
+-spec handle_list(wh_json:object(), whapps_call:call()) -> 'ok'.
+handle_list(Data, Call) ->
+    CallerIdNumber = whapps_call:caller_id_number(Call),
+    lager:debug("callerid number before this module: ~s ", [CallerIdNumber]),
+    CaptureGroup = whapps_call:kvs_fetch('cf_capture_group', Call),
+    lager:debug("capture_group ~s ", [CaptureGroup]),
+
+    << C1:8, C2:8, Dest/binary >> = CaptureGroup,
```

Comment:
`<<CIDKey:2/binary, Dest/binary>> = CaptureGroup` should match.

However, I'm concerned that you've limited the ability for JSON keys to be defined arbitrarily. I wonder if the capture group could allow you to define two capture groups in the regex, one for the Key and one for the Destination. You'll want to look in `cf_util:test_callflow_patterns/3` for that.

However, for this initial PR, don't worry about it. Just update the docs (thanks for those!) to reflect that the CID Key *must* be 2 digits.